src/contexts/Photos/Photos.context.tsx

```diff
@@ -131,12 +144,42 @@ export const PhotosContextProvider: React.FC = ({ children }) => {
     }),
   );
 
+  useEffect(() => {
+    checkShouldEnableSync()
```

Comment:
For future references and in order to avoid code duplication, you can pipe promises to avoid callback hells:

`checkShouldEnableSync()
      .then((enabled) => {
        return asyncStorageService.savePhotosSyncIsEnabled(enabled)
      })
       .then(() => setSyncEnabled(enabled))
      .catch((error) => {
        reportError(error);
      });`